src/access/profiling/parser.py

```diff
@@ -66,3 +73,32 @@ def _convert_from_string(value: str) -> Any:
         except Exception:
             continue
     return value
+
+
+def _read_text_file(file_path: Any) -> str:
+    """Checks whether file_path is a valid path to a text file and tries to read it.
+
+    Args:
+        file_path (Any): the object to check
+
+    Returns:
+        str: The text within the file.
+
+    Raises:
+        TypeError: if file_path is not a valid path
+        FileNotFoundError: if file_path is a path, but is not a file or doesn't exist.
+        ValueError: if file_path is a file, but cannot be read as a text file.
+    """
+
+    try:
+        path = Path(file_path)
+    except TypeError as e:
+        raise TypeError(f"{file_path} is not a valid path.") from e
+
+    if not path.is_file():
+        raise FileNotFoundError(f"{file_path} is not a file or doesn't exist.")
+
+    try:
+        return path.read_text()
+    except UnicodeDecodeError as e:
+        raise ValueError(f"{file_path} is not a text file.") from e
```

Comment:
Hmm - should this be `TypeError` instead of `ValueError` (because presumably a binary file was passed instead of a text file).